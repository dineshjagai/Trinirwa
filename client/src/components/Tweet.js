/* eslint-disable no-unused-vars */
import 'bootstrap/dist/css/bootstrap.css';
import React, { useEffect, useState } from 'react';
import Divider from '@material-ui/core/Divider';
import './tweet.css';
import LikeIcon from '@material-ui/icons/FavoriteBorderRounded';
import IconButton from '@material-ui/core/IconButton';
import CloseRoundedIcon from '@material-ui/icons/CloseRounded';
import TextField from '@material-ui/core/TextField';
import Tooltip from '@material-ui/core/Tooltip';
import CommentInput from './CommentInput';
import {
  getAvatar,
  updateTweetLikes,
  isLiked,
  unLikeTweet,
  likeTweet,
} from './Module';
import { getCurrentUsername } from '../auth/authServices';

export default function Tweet({ data, handleDelete }) {
  const user = getCurrentUsername();
  const [isLikedBool, setIsLike] = useState(false);
  const [likes, setLikes] = useState(data.tweet_likes);
  const [id] = useState(data.tweet_id);
  const [blocks, setBlocks] = useState(data.tweet_blocks);
  const [isOwner] = useState(data.user === user);
  const [avatar, setAvatar] = useState('');
  // eslint-disable-next-line eqeqeq
  const [isMedia] = useState(data.type == 'media');

  const getData = async () => {
    console.log('data.user:', data.user);
    await getAvatar(data.user).then((res) => {
      setAvatar(res.data.avatar[0].profile_picture);
    }).catch((err) => {
      console.log(err.message);
    });
    await isLiked(user, data.tweet_id).then((res) => {
      if (res.data.bool.length) {
        setIsLike(true);
      } else {
        setIsLike(false);
      }
    }).catch((err) => {
      console.log(err.message);
    });
  };
  const handleLike = () => {
    if (isLikedBool) {
      unLikeTweet(user, id).then((res) => {
        if (res.status === 200) {
          updateTweetLikes(id, likes - 1).then((result) => {
            console.log(result.message);
            setLikes(likes - 1);
          }).catch((err) => {
            console.log(err.message);
          });
        }
      }).catch((err) => {
        console.log(err.message);
      });
    } else {
      likeTweet(user, id).then((res) => {
        updateTweetLikes(id, likes + 1).then((result) => {
          console.log(result);
          setLikes(likes + 1);
        }).catch((err) => {
          console.log(err.message);
        });
      }).catch((err) => {
        console.log(err.message);
      });
    }
    setIsLike(!isLikedBool);
  };

  useEffect(() => {
    getData();
  }, []);
  const date = (data.tweet_date.split('T'))[0];
  const newAvatar = `/viewFile/${avatar}`;
  const newPicture = `/viewFile/${data.content}`;
  let newMediaTweet;
  if (isMedia) {
    newMediaTweet = <img className="tweet_image" id="tweet_media_picture" src={newPicture} alt="" />;
  } else {
    newMediaTweet = data.content;
  }

  return (
    <div id="container_tweet">
      <div className="tweet_header">
        <div className="tweet_img">
          <img className="tweet_header" id="tweet_author_picture" src={newAvatar} alt="" />
        </div>
        <div className="tweet_text">
          <span
            style={{
              fontSize: '13px',
              fontWeight: 'bold',
            }}
            id="author_username"
          >

            {data.user}
          </span>
          <span style={{ fontSize: '10px' }} id="date">
            {date}
          </span>
        </div>
      </div>
      <div id="menu">
        <Tooltip title={isOwner ? 'Delete post' : 'Hide post'} placement="top">
          <IconButton
            style={{
              borderRadius: '50%',
              padding: '3',
              color: '#0C8367',
            }}
            onClick={() => handleDelete(id, isOwner)}
          >
            <CloseRoundedIcon />
          </IconButton>
        </Tooltip>
      </div>
      <Divider variant="middle" />
      <div className="tweet_content">
        <p style={{ textAlign: 'left' }}>
          {newMediaTweet}
        </p>
      </div>
      <div className="comment">
        <CommentInput />
      </div>
      <div className="tweet_bottom">
        <div className="likes">
          <IconButton
            onClick={handleLike}
            style={{
              borderRadius: '50%',
              padding: '3',
              color: isLikedBool ? 'red' : 'black',
            }}
          >
            <LikeIcon />
          </IconButton>
          <span>{`${likes} likes`}</span>
        </div>
      </div>
    </div>
  );
}
