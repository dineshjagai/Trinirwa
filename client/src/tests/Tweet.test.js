import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import Tweet from '../components/Tweet';

test('Test Tweet snapshot', () => {
  const component = renderer.create(<Tweet />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
