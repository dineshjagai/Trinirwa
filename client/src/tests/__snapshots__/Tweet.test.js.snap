// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Tweet snapshot 1`] = `
<div
  id="container_tweet"
>
  <div
    className="tweet_header"
  >
    <div
      className="tweet_img"
    >
      <img
        alt=""
        className="tweet_header"
        id="tweet_author_picture"
        src="/viewFile/"
      />
    </div>
    <div
      className="tweet_text"
    >
      <span
        id="author_username"
        style={
          Object {
            "fontSize": "13px",
            "fontWeight": "bold",
          }
        }
      >
        isimbib
      </span>
      <span
        id="date"
        style={
          Object {
            "fontSize": "10px",
          }
        }
      >
        2008-09-07
      </span>
    </div>
  </div>
  <div
    id="menu"
  >
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "borderRadius": "50%",
          "color": "#0C8367",
          "padding": "3",
        }
      }
      tabIndex={0}
      title="Hide post"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18.3 5.71a.9959.9959 0 00-1.41 0L12 10.59 7.11 5.7a.9959.9959 0 00-1.41 0c-.39.39-.39 1.02 0 1.41L10.59 12 5.7 16.89c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L12 13.41l4.89 4.89c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L13.41 12l4.89-4.89c.38-.38.38-1.02 0-1.4z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>
  <hr
    className="MuiDivider-root MuiDivider-middle"
  />
  <div
    className="tweet_content"
  >
    <p
      style={
        Object {
          "textAlign": "left",
        }
      }
    >
      Hiii
    </p>
  </div>
  <div
    className="tweet_bottom"
  >
    <div
      className="likes"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "borderRadius": "50%",
            "color": "black",
            "padding": "3",
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19.66 3.99c-2.64-1.8-5.9-.96-7.66 1.1-1.76-2.06-5.02-2.91-7.66-1.1-1.4.96-2.28 2.58-2.34 4.29-.14 3.88 3.3 6.99 8.55 11.76l.1.09c.76.69 1.93.69 2.69-.01l.11-.1c5.25-4.76 8.68-7.87 8.55-11.75-.06-1.7-.94-3.32-2.34-4.28zM12.1 18.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
      <span>
        0 likes
      </span>
    </div>
  </div>
</div>
`;
